### DISCUSSION TITLE
levenshtein_at_thresholds("Description",2)

### DISCUSSION THREAD

**Discussion by msiemionCalistapw** [moj-analytical-services/splink#702] [Q&A] [answered]
what does the ",2" and ",[1,2]" mean in levenshtein_at_thresholds ?

**Comment by RobinL** [comment, marked as answer]
That will create one comparison level with a threshold distance of 1, and a second at a threshold distance of 2

There's some documentation here:
https://github.com/moj-analytical-services/splink/blob/5af45c54416994b2d85132d6a9f3e5149cc1bcd9/splink/comparison_library.py#L153

And the comparison has a human_readable_description property that will explain what it does:
https://github.com/moj-analytical-services/splink/blob/5af45c54416994b2d85132d6a9f3e5149cc1bcd9/splink/comparison.py#L467

So for instance:

```
from splink.duckdb.duckdb_comparison_library import (
    levenshtein_at_thresholds,
)

print(levenshtein_at_thresholds("my_col", [1,2,3]).human_readable_description)
```

will output

```
Comparison 'Exact match vs. levenshtein at thresholds 1, 2, 3 vs. anything else' of my_col.
Similarity is assessed using the following ComparisonLevels:
    - 'Null' with SQL rule: my_col_l IS NULL OR my_col_r IS NULL
    - 'Exact match' with SQL rule: my_col_l = my_col_r
    - 'levenshtein <= 1' with SQL rule: levenshtein(my_col_l, my_col_r) <= 1
    - 'levenshtein <= 2' with SQL rule: levenshtein(my_col_l, my_col_r) <= 2
    - 'levenshtein <= 3' with SQL rule: levenshtein(my_col_l, my_col_r) <= 3
    - 'All other comparisons' with SQL rule: ELSE
```
